### PR TITLE
deprecate `ZArith.Zdigits`

### PR DESCRIPTION
**Pull request by andres-erbsen**
- [x] Added **changelog**.

### CODE REVIEW

**Comment by andres-erbsen** [comment]
@coqbot run full ci

File: doc/changelog/11-standard-library/17733-deprecate-Zdigits.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Deprecated:** :g:`ZArith.Zdigits`
+  Describe your change here but do not end with a period
```
Comment (Alizter) [inline doc/changelog/11-standard-library/17733-deprecate-Zdigits.rst:2]: needs updating

**Review by Alizter — COMMENTED** [review]
could you add what to use instead for the deprecations? i.e. using the note option of the deprecated attribute.

**Comment by andres-erbsen** [comment]
> could you add what to use instead for the deprecations? i.e. using the note option of the deprecated attribute.

I added `note=` statements. The standard library does not have an exact drop-in replacement for this functionality, so some of the recommendations are based on anticipated context of use. I added a corresponding request for comments to the changelog.

**Comment by andres-erbsen** [comment]
Full CI failures at https://github.com/coq/coq/commit/19123b441c7ce6f54ac7b45d041e9898fdd83f3f look unrelated (geocoq and ltac2 compiler)

**Comment by gares** [comment]
ping @coq/number-maintainers this PR seems ready, can you review it quickly please?

**Review by proux01 — APPROVED** [review]
Sorry, I thought I already merged that one (I apparently mixed it with https://github.com/coq/coq/pull/17601 ), let's merge.

**Comment by proux01** [comment]
@coqbot merge now

**Pull request merged**